emerge-gitclone

```diff
@@ -9,17 +9,36 @@ import sys
 
 import portage
 
-branch = 'master'
+# Find the version of the currently running release.
+release = False
 with open('/usr/share/coreos/release') as release_file:
 	for line in release_file:
 		if line.startswith('COREOS_RELEASE_VERSION='):
-			branch = 'build-' + line.replace('.', '=', 1).split('=', 2)[1]
+			release = line.split('=', 1)[1].strip()
+
+# Attempt to read Git commits from this release's manifest.
+branch = 'master'
+commits = {}
+if release:
+	branch = 'build-' + release.split('.', 1)[0]
+	manifest = "https://raw.githubusercontent.com/coreos/manifest/v%s/release.xml" % release
+	try:
+		from xml.dom.minidom import parseString as pxs
+		try:  # Python 3
+			from urllib import request as req
+		except:  # Python 2
+			import urllib2 as req
+		for repo in pxs(req.urlopen(manifest).read()).getElementsByTagName('project'):
+			commits[repo.getAttribute('name')] = repo.getAttribute('revision')
+	except:
```

Comment:
and this is conventionally `except Exception` to avoid catching async exceptions like the one from SIGINT.